types/src/trusted_state.rs

```diff
@@ -167,7 +160,12 @@ impl TrustedState {
 
             // Verify the latest ledger info inside the latest epoch.
             let new_verifier = VerifierType::TrustedVerifier(new_epoch_info);
-            new_verifier.verify(latest_li)?;
+
+            // If these are the same, then we do not have a LI for the next Epoch and hence there
+            // is nothing to verify.
+            if epoch_change_li != latest_li {
+                new_verifier.verify(latest_li)?;
+            }
```

Comment:
IIRC this case should never happen for a client requesting a state proof from a server.
If the server is exactly on an epoch boundary (i.e., its latest_li has next_validator_set = Some(...)), it won't respond with that epoch change in the validator change proof. What is the use case here then?

Reply:
Why is this the case? Is there an explicit rule that says do not return an LI that has a next_validator_set somewhere in the code?

Reply:
see this comment chain in the original `TrustedState` PR: https://github.com/libra/libra/pull/2700#discussion_r386530726

Reply:
or the storage logic: https://github.com/libra/libra/blob/master/storage/libradb/src/lib.rs#L236

Reply:
Ahh, how interesting. When I went to the libradb function: https://github.com/libra/libra/blob/master/storage/libradb/src/lib.rs#L759

which seemed to return the appropriate ValidatorChangeProof -- or at least my tests worked and I ended copying that to the latest li.